pkg/controller/csi_extension_server.go

```diff
@@ -0,0 +1,128 @@
+package controller
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	vgsext "github.com/dell/dell-csi-extensions/volumeGroupSnapshot"
+	"github.com/dell/gopowerstore"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const StateReady = "Ready"
+
+func (s *Service) CreateVolumeGroupSnapshot(ctx context.Context, request *vgsext.CreateVolumeGroupSnapshotRequest) (*vgsext.CreateVolumeGroupSnapshotResponse, error) {
+	log.Infof("CreateVolumeGroupSnapshot called with req: %v", request)
+
+	err := validateCreateVGSreq(request)
+	if err != nil {
+		log.Errorf("Error from CreateVolumeGroupSnapshot: %v ", err)
+		return nil, err
+	}
+	var reqParams gopowerstore.VolumeGroupSnapshotCreate
+	reqParams.Name = request.GetName()
+	reqParams.Description = request.GetDescription()
+	arr := strings.Split(request.SourceVolumeIDs[0], "/")[1]
```

Comment:
check if Split() returns 2 elements before accessing the second element

Reply:
done